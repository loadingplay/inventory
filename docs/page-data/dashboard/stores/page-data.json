{"componentChunkName":"component---src-pages-dashboard-stores-index-jsx","path":"/dashboard/stores/","webpackCompilationHash":"2779579faed300d22aca","result":{"pageContext":{"isCreatedByStatefulCreatePages":true}}}