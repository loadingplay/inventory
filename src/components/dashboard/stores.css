.stores_wrapper {
  margin-top: 10px;
}
header {
  font-weight: 700;
  text-transform: capitalize;
  background: #33b5e5;
  color: #fff;
  padding: 5px 10px;
}

.stores_wrapper ul {
  list-style-type: none;
  margin: 0;
  padding: 0;
  width: 200px;
  background-color: #f1f1f1;
}

.stores_wrapper li a {
  display: block;
  color: #000;
  padding: 8px 16px;
  text-decoration: none;
}

/* Change the link color on hover */
.stores_wrapper li a:hover {
  background-color: rgb(150, 150, 150);
  color: white;
}

.table {
  background: #ffffff;
}

tr.stores_selected {
  background: rgba(0,0,0,.075);
}

.table-info > th {
  background-color: #33b5e5;
  color: #ffffff;
}

.search_actions {
  display: grid;
  grid-template-columns: 1fr 1fr;
}

button {
  width: 240px;
  margin-bottom: 10px;
}

.list_btn > div {
  text-align: right;
}
