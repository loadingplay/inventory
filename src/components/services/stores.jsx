export default class StoreLoader {
  constructor(access_token) {
    this.access_token = access_token;
  }

  loadStoreList = async (page, cellar_id) => {
    let response, json_data;

    response = await fetch(
      `https://replenishments.loadingplay.com/replenishment?items=100&page=${page}&cellar_id=${cellar_id}`,
      {
        "headers": {
          "Authorization": `Bearer ${this.access_token}`
        }
      });

    json_data = await response.json();
    return json_data;
  }
}
