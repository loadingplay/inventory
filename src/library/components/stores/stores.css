.stores_selected, .stores_selected:hover {
    background-color: #333333;
    color: #eeeeee;
}
