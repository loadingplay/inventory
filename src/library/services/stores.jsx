export class StoreLoader {
  constructor(access_token) {
    this.access_token = access_token;
  }

  loadCellars = async () => {
    let response,
      json_data;

    response = await fetch("https://apibodegas.loadingplay.com/v1/cellar", {
      headers: {
        "Authorization": `Bearer ${this.access_token}`
      }
    });
    json_data = await response.json();
    return json_data;
  }

  loadProducts = async (page, cellar_id, sku_filter) => {
    let response, json_data;

    response = await fetch(
      `https://replenishments.loadingplay.com/replenishment?items=100&page=${page}&cellar_id=${cellar_id}&search_word=${sku_filter}`,
      {
        "headers": {
          "Authorization": `Bearer ${this.access_token}`
        }
      });

    json_data = await response.json();
    return json_data;
  }

  setSelectedCellar = (cellar) => {
    localStorage.setItem("selected-cellar", JSON.stringify(cellar));
  }

  getSelectedCellar = () => {
    return JSON.parse(localStorage.getItem("selected-cellar"));
  }
}
