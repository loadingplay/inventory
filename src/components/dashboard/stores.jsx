import React, { useEffect, useState } from 'react'
import './stores.css'
import { Link } from 'gatsby'
import authService from '../../services/authService'
import cellarService from '../../services/cellarService'

const stores = (props) => {

    const [data, dataSet] = useState([]);
    const [addCellarIsClicked, setAddCellarIsClicked] = useState(false);
    const [deleteCellarIsClicked, setDeleteCellarIsClicked] = useState(false);
    const [cellarChecked, setCellarChecked] = useState(false);
    const [cellarId, setCellarId] = useState(0);
 
    async function fetchData(){
      let token = authService.getCurrentToken();
      let allCellars = await cellarService.getAllCellar(token);
      console.log(allCellars)
      dataSet(allCellars); 
    }

    useEffect(() => {
      fetchData();
    }, [])


    function handleCheckCellar(event){
      const value = event.target.value;
      const checked = event.currentTarget.checked;
      setCellarChecked(checked)
      setCellarId(value)
    }

    async function handleAddCellarClick(){
      setAddCellarIsClicked(true)
      console.log(addCellarIsClicked)
      let objPost = {
        name: 'Test Post Nuevo',
        description: 'Description Nueva'
      }
      let token = authService.getCurrentToken();
      await cellarService.postCellar(objPost, token)
      fetchData();
    }

    async function handleDeleteCellarClick(){
      setDeleteCellarIsClicked(true)
      let token = authService.getCurrentToken();
      await cellarService.deleteCellar(cellarId, token);
      fetchData();
    }

    return (
      <section className="stores_wrapper">
        <table className="table table-hover table-borderless">
          <thead>
            <tr className="table-info">
              <th scope="col">tiendas</th>
            </tr>
          </thead>
          <tbody>
          {data.cellars ? data.cellars.map((item, i) => 
            <tr key={i}>
               <td><input type="checkbox" name="cellar" value={item.id} checked={cellarChecked} onChange={handleCheckCellar}></input> {item.name}</td>
            </tr>
          ): <tr><td>Buscando...</td></tr>}
          </tbody>
        </table>
        <div>
          <button type="button" className="btn btn-success stores_wrapper_button" onClick={handleAddCellarClick}>
            AÃ±adir Bodega
          </button> 
          <button type="button" disabled={cellarId == 0} className="btn btn-danger stores_wrapper_button" onClick={handleDeleteCellarClick}>
            Eliminar Bodega
          </button>
        </div>
      </section>
    )
}

export default stores;