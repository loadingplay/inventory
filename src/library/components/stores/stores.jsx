import React, { Component } from 'react'
import { StoreLoader } from '../../services';
import './stores.css'

export class Stores extends Component {

  constructor(props) {
    super(props);
    this.state = {
      cellars: null,
      selected_id: 0
    };

    this.store_api = new StoreLoader("");
  }

  loadCellars = async (token) => {
    let json_data;

    this.store_api.access_token = token
    json_data = await this.store_api.loadCellars();

    this.setState({
      "cellars": json_data.cellars
    });
  }

  saveSelectedCellar = (cellar) => {
    this.store_api.setSelectedCellar(cellar);
  }

  componentWillReceiveProps(newProps) {
    if (newProps.access_token !== "")
      this.loadCellars(newProps.access_token);
  }

  render() {
    let items = this.state.cellars ? this.state.cellars.map((item, index) => {
      return (
        <tr
          key={index}
          className={this.state.selected_id === item.id ? "stores_selected" : ""}
        >
          <td
            onClick={() =>Â {
              this.setState({ selected_id: item.id });
              this.saveSelectedCellar(item);
              this.props.cellarSelected(item.id);
            }}
          >
            {item.name}
          </td>
        </tr>);
    }) : (<tr><td>Cargando...</td></tr>);

    return (
      <section className="stores_wrapper">
        <table className="table table-hover table-borderless table-sm">
          <thead>
            <tr className="table-info">
              <th scope="col">tiendas</th>
            </tr>
          </thead>
          <tbody>
            {items}
          </tbody>
        </table>
      </section>
    )
  }
}
