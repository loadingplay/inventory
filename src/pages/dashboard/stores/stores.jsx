import React, { Component } from 'react'
import { StoreLoader } from '../../../library/services';
import './stores.css'

export class Stores extends Component {

  constructor(props) {
    super(props);
    this.state = {
      cellars: null,
      selected_id: 0
    };
  }

  loadCellars = async (token) => {
    let store_api,
      json_data;

    store_api = new StoreLoader(token);
    json_data = await store_api.loadCellars();

    this.setState({
      "cellars": json_data.cellars
    });
  }

  componentWillReceiveProps(newProps) {
    if (newProps.access_token !== "")
      this.loadCellars(newProps.access_token);
  }

  render() {
    let items = this.state.cellars ? this.state.cellars.map((item, index) => {
      return (
        <tr
          key={index}
          className={this.state.selected_id === item.id ? "stores_selected" : ""}
        >
          <td
            onClick={() =>Â {
              this.setState({ selected_id: item.id });
              this.props.cellarSelected(item.id);
            }}
          >
            {item.name}
          </td>
        </tr>);
    }) : (<tr><td>Cargando...</td></tr>);

    return (
      <section className="stores_wrapper">
        <table className="table table-hover table-borderless table-sm">
          <thead>
            <tr className="table-info">
              <th scope="col">tiendas</th>
            </tr>
          </thead>
          <tbody>
            {items}
          </tbody>
        </table>
      </section>
    )
  }
}
