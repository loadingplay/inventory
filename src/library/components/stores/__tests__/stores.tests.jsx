import React from "react";
import { Stores } from "../stores";
import { shallow } from "enzyme";


describe("Stores", () => {
  test("it should reder", () => {
    const wrapper = shallow(
      <Stores></Stores>
    );

    expect(wrapper.instance().load_status).toEqual("idle");
    expect(wrapper).toMatchSnapshot("default");
  });

  test("it should try to load cellars", () => {
    const wrapper = shallow(
      <Stores />
    );

    wrapper.instance().loadCellars = jest.fn().mockResolvedValue({});
    wrapper.setProps({
      accessToken: "test-token"
    });

    expect(wrapper.instance().loadCellars).toHaveBeenCalledTimes(1);
    expect(wrapper.instance().loadCellars).toBeCalledWith("test-token");
  });

  test("it should call load cellars from api", async () => {
    let promise;
    const wrapper = shallow(<Stores onHQCellarLoaded={jest.fn().mockReturnThis()} />);

    wrapper.instance().store_api.loadCellars = jest.fn().mockResolvedValue({
      cellars: [
        {
          id: 1,
          for_sale: false
        },{
          id: 2,
          for_sale: true
        }
      ]
    });

    expect(wrapper.instance().load_status).toEqual("idle");
    promise = wrapper.instance().loadCellars("test-token");
    expect(wrapper.instance().load_status).toEqual("loading");
    await promise;
    expect(wrapper.instance().props.onHQCellarLoaded).toHaveBeenCalledWith(2);
    expect(wrapper.instance().load_status).toEqual("loaded");
  });
});
