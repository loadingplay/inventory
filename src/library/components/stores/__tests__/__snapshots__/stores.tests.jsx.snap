// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stores it should reder: default 1`] = `
<section
  className="stores_wrapper"
>
  <table
    className="table table-hover table-borderless table-sm"
  >
    <thead>
      <tr
        className="table-info"
      >
        <th
          scope="col"
        >
          tiendas
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Cargando...
        </td>
      </tr>
    </tbody>
  </table>
</section>
`;
